artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/AMQPConnectionContext.java

```diff
@@ -87,9 +89,8 @@
 
    private final boolean useCoreSubscriptionNaming;
 
-   private boolean isSchedulingCancelled;
-   private ScheduledFuture scheduledFuture;
-   private final Object schedulingLock = new Object();
+   private static final FutureTask<Void> voidFuture = new FutureTask<>(() -> { }, null);
```

Comment:
VOID_FUTURE as this is a constant.

Reply:
on that note what is its purpose? surely can just set the reference to null

Reply:
ScheduledFutureRef is used for 2 purpose: keeping the last scheduledFuture and the scheduling status, so I need tree states for scheduledFutureRef:
- VOID_FUTURE before onRemoteOpen / NOT SCHEDULED / SCHEDULING ALLOWED
- NULL after close / SCHEDULING NOT ALLOWED
- ELSE after onRemoteOpen and before close / SCHEDULED / SCHEDULING ALLOWED

Reply:
I suppose that @michaelandrepearce means to rename the variable from `voidFuture` to `VOID_FUTURE`

Reply:
no i was literally meaning why the need for this, just set reference to null, i dont see the need for three states, two states is all thats needed looking at code, and the state changes.

my first comment was just it should be capitlised, still not changed, but that was before i looked at its use and now question its need at all.

Reply:
ok, think ive spotted what you're trying to achieve with the VOID_FUTURE. still part about capitilised naming for constant remains.